### ISSUE TITLE
Comments: Create new comment details view

### ISSUE THREAD

**Issue by dvdchr** [wordpress-mobile/WordPress-iOS#17087]
Ref: pbArwn-2ks-p2

## Summary
Create a new Comment Details view, which will be accessible from the Comments and Notifications page. Note that some components will be addressed separately through different issues, and some functionalities may depend on ongoing development. Here's the design from Zeplin:

![Screen Shot 2021-08-26 at 16 08 28](https://user-images.githubusercontent.com/1299411/130935920-d5e1d4c6-22eb-4085-a46c-b1a3f8b70fc8.png)

Some notes regarding components marked red (going from the top):
- The edit button will be visually included in this issue, but the functionality depends on #17000, and _may_ be wired later on. When tapped, the new edit comment view should be shown in full modal presentation.
  - Since #17046 is merged, the functionality can be included here, but I'll need to make sure that the edit view feature flag is also enabled.
- Same as above, the reply button will be visually included in this issue, but the functionality will be addressed through a separate issue.
- Lastly, the moderation bar will completely be excluded from this and will be addressed through a different issue.

### Site Comments

- [x] Add a feature flag for comment detail. #17088
- [x] Create the initial view using Swift. #17088
- [x] Comment header cell. #17095
- [x] Create a generic cell for the other similar-looking fields (e.g. web address, email address, IP address). #17105 
- [x] Comment reply indicator cell. #17106 
- [x] #17108
- [x] Add "Delete Permanently" button when the Comment is in Trashed state. #17286
- [x] Apply a visual blur effect when the page is scrolled. #17255
- [x] Adjust certain cells' visibility based on account permission. #17147
- [x] https://github.com/wordpress-mobile/WordPressKit-iOS/pull/463
- [x] #17399
- [x] #17408
- [x] Fetch the parent Comment object if `parentID` exists, to get the replied author's name (i.e. "Reply to `Pamela Nguyen`").  #17246
- [x] Show the new details view from My Site > Comments. #17088

#### Finishing touches
- [x] Review accessibility.
- [x] Review Tracks.
- [x] #17466

#### Issues
- [x] Don't show the web address cell if the content is empty (discussion [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17105#issuecomment-912662400)). Addressed in #17147.
- [x] Backport `arrowshape.turn.up.backward` symbol, as it's only available in iOS 14 and above. More details [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17147#issuecomment-925169160). Addressed in #17469.
- [x] #17452
- [x] #17451
- [x] #17460
- [x] #17463
- [x] #17462
- [x] Fix comment content header not resized properly when using accessibility text size. Addressed in #17464.
- [x] Fix comment content Reply and Like buttons not resized properly when using accessibility text size. Addressed in #17464.

---

### Notification Comments

- [ ] Refactor notification list to show detail using custom view controllers while maintaining pagination.
- [ ] Refactor Comment Details so it could work with a `Notification` object.
- [ ] Fetch and populate the data required for comment details, when coming from Notifications. May require changes to WordPressKit.
- [ ] Show the new details view from Notifications.

#### Finishing touches
- [ ] Review accessibility.
- [ ] Review Tracks.
- [ ] Enable the feature flag for public and update release notes.

### Other non-blocking issues
- [ ] Address transition issue when tapping on the Comment header cell, as described in [this comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/17095#discussion_r698784435). This is a legacy issue.
- [ ] Fix an issue where the back button's label on the navigation bar is gone after entering split view, as described in [this comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/17105#issuecomment-910615192). This is a legacy issue.
- [ ] **[Low priority]** When the orientation change is spammed, content height calculation may get into a state where the web view height is higher than the actual content. Discussion [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17129#pullrequestreview-749499061). 
- [ ] Formatted comment contents from self-hosted sites are not rendered with the correct spacing. More details [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17129#pullrequestreview-749499061) and [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17129#issuecomment-916029042).
- [ ] Fix VoiceOver pathing resets to the top of the screen after reading web view contents. More details [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17464).
